src/adapter/src/coord/appends.rs

```diff
@@ -131,6 +131,17 @@ macro_rules! guard_write_critical_section {
 }
 
 impl Coordinator {
+    /// Send a message to the Coordinate to start a group commit.
+    pub(crate) fn trigger_group_commit(&mut self) {
+        self.internal_cmd_tx
+            .send(Message::GroupCommitInitiate)
+            .expect("sending to self.internal_cmd_tx cannot fail");
+        // Avoid excessive `Message::GroupCommitInitiate` by resetting the periodic table
+        // advancement. The group commit triggered by the message above will already advance all
```

Comment:
If the group commit action does this, should we instead reset the timer there?

Reply:
As in at the top of `try_group_commit`?

Reply:
Or whenever we receive the `Message::GroupCommitInitiate` message in the message handler?

Reply:
I think at the top of `group_commit_initiate`? That's the thing that is doing the full table advancement, so it could also know it can reset the tick.

Reply:
Either way I thought about that actually. I think the pro of that approach is that it also lowers the amount of calls to append when we have to block and wait for the system clock to catch up to the timestamp oracle clock. The con is that you end up with the following happy path which is a bit weird:
1. Interval ticks, which sends a `Message::GroupCommitInitiate` and causes the interval to implicitly reset.
2. We start the group commit and reset the interval again.

It's not the worst and might even be worth it, I just thought that the double reset was a little weird.

Reply:
Ah, this is a good point. Ok, I think we should leave as is then.

Reply:
Though the following scenario still exists which is a bit of a bummer:
1. Timer kicks off a table advancement.
2. The timer restarts.
3. The system clock is ahead of the timestamp oracle clock so we start an async task that kicks off a table advancement in less than or equal to 1 second.

So again we have have a double table advancement in the same second. It's much less likely but it's still there, and would be prevented by moving the reset to `group_commit_initiate` or `try_group_commit`.